### DISCUSSION TITLE
Dark Mode Selector for Base Styles plugin

### DISCUSSION THREAD

**Discussion by graydenshand** [tailwindlabs/tailwindcss#3551] [Help] [answered]
I'm adding base styles to a site as follows

```
// tailwind.config.js

...
plugins:
  plugin(function({ addBase, theme }) {
    a: {
      fontWeight: theme('fontWeight.medium'),
      color: theme('colors.purple.700'),
      textDecoration: theme('textDecoration.underline'),
      '&:hover': {
        color: theme('colors.purple.600'),
      },
      // dark_mode_selector ?
    }

 }
```

Basically, I'm looking to specify dark-mode base styles in a way analogous to the way I am specifying `'&:hover'`.

Is this possible?

**Comment by Exotelis** [comment, marked as answer]
Hey @graydenshand

at least for `darkMode` is `media` you could do it like this:

```js
plugin(function ({ addBase, theme }) {
  const darkAnchor = {
    backgroundColor: theme('colors.gray.100'),
  };

  addBase({
    a: {
      backgroundColor: theme('colors.gray.900'),
    },
    '.dark': {
      a: darkAnchor,
    },
    '@media (prefers-color-scheme: dark)': {
      a: darkAnchor,
    },
  });
}),
```

Unfortunately, the `.dark` class definition will be removed if the class is not set initially.

However, if you implement a toggle functionality such as https://tailwindcss.com/docs/dark-mode#toggling-dark-mode-manually I guess tailwind would also parse the line `document.documentElement.classList.add('dark')` and `.dark` would not be removed from your definition ;)

Update: I just double checked it. As long as `document.documentElement.classList.add('dark')` is in some file that is parsed by tailwind, it would not remove the `.dark` definition ;)
